torch/ao/ns/fx/qconfig_multi_mapping.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
+
+import copy
+from typing import Any, Callable, Dict, List, Union
+
+import torch
+from torch.ao.quantization import QConfigMapping
+from torch.ao.quantization.qconfig import QConfigAny
+
+_QCONFIG_STYLE_ORDER: List[str] = [
+    "global_qconfig",
+    "object_type_qconfigs",
+    "module_name_regex_qconfigs",
+    "module_name_qconfigs",
+    "module_name_object_type_order_qconfigs",
+]
+
+_QCONFIG_STYLE_TO_METHOD: Dict[str, str] = {
+    "global_qconfig": "set_global",
+    "object_type_qconfigs": "set_object_type",
+    "module_name_regex_qconfigs": "set_module_name_regex",
+    "module_name_qconfigs": "set_module_name",
+    "module_name_object_type_order_qconfigs": "set_module_name_object_type_order",
+}
+
+
+class QConfigMultiMapping:
```

Comment:
any thoughts on how the user should think about the order of the results?  For example, if the user does

```
QConfigMultiMapping().\
    set_global([a, b]).\
    set_object_type(nn.Conv2d, [c, d])
```

how would the results be ordered in the resulting loggers?  I think any reasonable order is fine, I'm just asking what the actual order would be.  We should probably document it.

Reply:
i'd probably rather split this consideration into a seperate PR because if we're going to have guarantees about ordering we'd probably want a more robust test suite for it. There's no explicit ordering considerations at the moment, thoguh I assume it'd be ordered as one would expected, i.e. fifo for qconfigs.

Reply:
makes sense, it seems that for anything other than testing one qconfig an order would be required in order to interpret the results correctly